frontend/src/tests/instructor-permission-test.js

```diff
@@ -15,14 +15,167 @@ export function instructorsPermissionTests(api) {
+
+    /**
+     * Switches the current active user to a user with only the instructor role.
+     * This function uses debug route to achieve user switching.
+     *
+     * @returns {Promise<void>}
+     */
+    async function switchToInstructorOnlyUser() {
+        let respSwitchToInstOnlyUser = await apiPOST(
+            `/debug/active_user`,
+            instructorOnlyUser
+        );
+        expect(respSwitchToInstOnlyUser).toHaveStatus("success");
+    }
+
+    /**
+     * Restores the active user to the default user (the user logged during test setup in beforeAll).
+     *
+     * @returns {Promise<void>}
+     */
+    async function restoreDefaultUser() {
+        let respSwitchBackUser = await apiPOST(
+            `/debug/active_user`,
+            defaultUser
+        );
+        expect(respSwitchBackUser).toHaveStatus("success");
+    }
 
     beforeAll(async () => {
         await databaseSeeder.seed(api);
         await databaseSeeder.seedForInstructors(api);
         session = databaseSeeder.seededData.session;
+
+        // default user should have both admin and instructor roles
+        let resp = await apiGET(`/debug/active_user`);
+        expect(resp).toHaveStatus("success");
+        expect(resp.payload.roles).toEqual(
+            expect.arrayContaining(["instructor", "admin"])
+        );
+        defaultUser = resp.payload;
+
+        resp = await apiGET(`/admin/sessions/${session.id}/contract_templates`);
+        expect(resp).toHaveStatus("success");
+        expect(resp.payload).toEqual(
+            expect.arrayContaining([expect.anything()])
+        );
+        existingContractTemplateId = resp.payload[0].id;
+
+        const instructorOnlyUserData = {
+            utorid: "instructor_only_test_user_utorid",
+            roles: ["instructor"],
+        };
+
+        resp = await apiPOST(`/debug/users`, instructorOnlyUserData);
+        expect(resp).toHaveStatus("success");
+
+        resp = await apiGET(`/debug/users`);
+        expect(resp).toHaveStatus("success");
+
+        resp.payload.forEach((user) => {
+            if (user.utorid === instructorOnlyUserData.utorid) {
+                expect(user.roles).toEqual(["instructor"]);
+                instructorOnlyUser = user;
+            }
+        });
+
+        expect(instructorOnlyUser).toBeDefined();
     }, 30000);
 
+    it("assigning to be instructor of a course grants instructor role", async () => {
+        await restoreDefaultUser();
+        const emptyRoleUserUtorid =
+            "course_assign_grant_instructor_test_user_utorid";
+        const emptyRoleUserData = {
+            utorid: emptyRoleUserUtorid,
+            roles: [],
+        };
+
+        let resp = await apiPOST(`/debug/users`, emptyRoleUserData);
+        expect(resp).toHaveStatus("success");
+
+        resp = await apiGET(`/debug/users`);
+        expect(resp).toHaveStatus("success");
+
+        let emptyRoleUserExists = false;
+        resp.payload.forEach((user) => {
```

Comment:
`resp.payload.find((user) => user.utorid === emptyRoleUserData.utorid)` will be null if nothing is found, otherwise it will be the user object. Alternatively, you could use `.some` to get a boolean.